R/select_nse.R

```diff
@@ -0,0 +1,509 @@
+# Code adapted from {poorman} by Nathan Eastwood [License: MIT]
+# https://github.com/nathaneastwood/poorman/blob/master/R/select_positions.R
+
+.select_nse <- function(select, data, exclude, ignore_case, regex = FALSE,
+                        verbose = FALSE) {
+  .check_data(data)
+  columns <- colnames(data)
+
+  # avoid conflicts
+  conflicting_packages <- .conflicting_packages("poorman")
+  on.exit(.attach_packages(conflicting_packages))
```

Comment:
Can you please explain how do we conflict _poorman_? 

AFAICT, there are no similarly named functions across _datawizard_ and _poorman_.

Reply:
There's a conflict with the select helpers. `poorman` exports `starts_with()` and friends, which create a conflict when we use our (unexported) select helpers. If I remove these two lines, it breaks some cases:
``` r
suppressPackageStartupMessages({
  library(datawizard)
  library(dplyr)
})

test_fun1 <- function(data, i) {
  get_columns(data, select = i)
}

test_fun1(iris, starts_with("Sep"))
#>     Sepal.Length Sepal.Width
#> 1            5.1         3.5
#> 2            4.9         3.0
#> 3            4.7         3.2
#> 4            4.6         3.1
#> 5            5.0         3.6
```
The reason why we don't have to do this for packages like `dplyr` is that `dplyr::starts_with()` errors while `poorman::starts_with()` doesn't. We actually use the error message of `dplyr` in some places of the code to get the select helper that was called

Reply:
I see; makes sense to anticipate this scenario then. Thanks for the explanation.